internal/parser/grammar.md

```diff
@@ -63,10 +63,14 @@ The parser directly implements the following grammar.
     MemberBracketExpression        = "[" string "]" .
     CallExpression                 = "(" ParameterList ")" .
     IndexExpression                = "[" Expression "]" .
+    StringExpression               = "\"" [ StringExpressionPart ] "\"" .
+    StringExpressionPart           = text | StringExpressionBlock .
+    StringExpressionBlock          = "${" Expression "}" .
     PrimaryExpression              = identifer
                                    | int_lit
                                    | float_lit
                                    | string_lit
+                                   | StringLiteralExpression
```

Comment:
Should this say `StringExpression`?

Reply:
Probably. I went back and forth on the name a few times.